src/coreclr/jit/flowgraph.cpp

```diff
@@ -52,6 +52,254 @@ static bool blockNeedsGCPoll(BasicBlock* block)
     return blockMayNeedGCPoll;
 }
 
+//------------------------------------------------------------------------------
+// fgInsertClsInitChecks : Wraps static init helper calls with "is class statically initialized"
+//                         inlined checks.
+//
+// Returns:
+//    PhaseStatus indicating what, if anything, was changed.
+//
+
+PhaseStatus Compiler::fgInsertClsInitChecks()
+{
+    if (!opts.OptimizationEnabled() || opts.IsReadyToRun())
```

Comment:
`opts.IsReadyToRun()` check should be unnecessary. It is covered by `fgHasOptStaticInit` condition below.